File: tmp-mei/fgenjs5030319945734616766/base/StrMap_1.js
import { base$$LinkedLens_2 } from "../base/LinkedLens_2.js";
import * as rt from "../rt/main.js";

export class base$$StrMap_1 extends base$$LinkedLens_2 {
  static $self = new base$$StrMap_1();

  get$imm(k_m$) {
  return base$$LinkedLens_2.get$imm$fun(k_m$, this);
}


get$mut(k_m$) {
  return base$$LinkedLens_2.get$mut$fun(k_m$, this);
}


get$read(k_m$) {
  return base$$LinkedLens_2.get$read$fun(k_m$, this);
}


keyEq$read(k1_m$, k2_m$) {
  return base$$StrMap_1.keyEq$read$fun(k1_m$, k2_m$, this);
}


isEmpty$read() {
  return base$$Map_2.isEmpty$read$fun(this);
}


map$mut(fImm_m$, fMut_m$, fRead_m$) {
  return base$$LinkedLens_2.map$mut$fun(fImm_m$, fMut_m$, fRead_m$, this);
}


put$imm(k_m$, v_m$) {
  return base$$LinkedLens_2.put$imm$fun(k_m$, v_m$, this);
}


put$mut(k_m$, v_m$) {
  return base$$LinkedLens_2.put$mut$fun(k_m$, v_m$, this);
}


put$read(k_m$, v_m$) {
  return base$$LinkedLens_2.put$read$fun(k_m$, v_m$, this);
}


map$imm(fImm_m$, fRead_m$) {
  return base$$LinkedLens_2.map$imm$fun(fImm_m$, fRead_m$, this);
}


map$read(fImm_m$, fRead_m$) {
  return base$$LinkedLens_2.map$read$fun(fImm_m$, fRead_m$, this);
}

  static keyEq$read$fun(k1_m$, k2_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}

File: tmp-mei/fgen4255027576711118260/base/StrMap_1.java
package base;
public interface StrMap_1 extends base.LinkedLens_2{
StrMap_1 $self = new StrMap_1Impl();
base.Opt_1 get$imm(Object k_m$);

base.Opt_1 get$mut(Object k_m$);

base.Opt_1 get$read(Object k_m$);

base.Bool_0 keyEq$read(Object k1_m$, Object k2_m$);

base.Bool_0 isEmpty$read();

base.LinkedLens_2 map$mut(base.MapMapImm_3 fImm_m$, base.MapMapMut_3 fMut_m$, base.MapMapRead_3 fRead_m$);

base.LinkedLens_2 put$imm(Object k_m$, Object v_m$);

base.LinkedLens_2 put$mut(Object k_m$, Object v_m$);

base.LinkedLens_2 put$read(Object k_m$, Object v_m$);

base.LinkedLens_2 map$imm(base.MapMapImm_3 fImm_m$, base.MapMapRead_3 fRead_m$);

base.LinkedLens_2 map$read(base.MapMapImm_3 fImm_m$, base.MapMapRead_3 fRead_m$);
static base.Bool_0 keyEq$read$fun(rt.Str k1_m$, rt.Str k2_m$, base.StrMap_1 $this) {
  return ((base.Bool_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Bool_0) null;
}})
);
}
}